src/pages/PlanPage/subRoutes/UpgradePlanPage/UpgradeForm2/Controllers/ProPlanController/ProPlanController.spec.tsx

```diff
@@ -0,0 +1,325 @@
+        if (planValue === Plans.USERS_BASIC) {
+          return res(ctx.status(200), ctx.json(mockAccountDetailsBasic))
+        } else if (planValue === Plans.USERS_PR_INAPPM) {
+          return res(ctx.status(200), ctx.json(mockAccountDetailsProMonthly))
+        } else if (planValue === Plans.USERS_PR_INAPPY) {
+          return res(ctx.status(200), ctx.json(mockAccountDetailsProYearly))
+        } else if (planValue === Plans.USERS_TRIAL) {
+          return res(ctx.status(200), ctx.json(mockAccountDetailsTrial))
+        }
+      }),
+      rest.patch(
+        '/internal/gh/codecov/account-details/',
+        async (req, res, ctx) => {
+          return res(ctx.status(200), ctx.json({ success: false }))
+        }
+      ),
+      graphql.query('GetAvailablePlans', (req, res, ctx) => {
+        return res(
+          ctx.status(200),
+          ctx.data({
+            owner: {
+              availablePlans: [basicPlan, proPlanMonth, proPlanYear, trialPlan],
+            },
+          })
+        )
+      }),
+      graphql.query('GetPlanData', (req, res, ctx) => {
+        return res(
+          ctx.status(200),
+          ctx.data({
+            owner: {
+              plan: { ...mockPlanDataResponse, trialStatus },
+            },
+          })
+        )
+      })
+    )
+
+    return { addNotification, user }
+  }
+
+  describe('when rendered', () => {
+    describe('when the user has a pro plan monthly', () => {
+      const props = {
+        setFormValue: jest.fn(),
+        register: jest.fn(),
+        newPlan: Plans.USERS_PR_INAPPM,
+        seats: 10,
+        errors: { seats: { message: '' } },
+      }
+      it('renders monthly option button', async () => {
+        setup({ planValue: Plans.USERS_PR_INAPPM })
+        render(<ProPlanController {...props} />, { wrapper: wrapper() })
+
+        const optionBtn = await screen.findByRole('button', { name: 'Monthly' })
+        expect(optionBtn).toBeInTheDocument()
+      })
+
+      it('renders annual option button', async () => {
+        setup({ planValue: Plans.USERS_PR_INAPPM })
+        render(<ProPlanController {...props} />, { wrapper: wrapper() })
+
+        const optionBtn = await screen.findByRole('button', { name: 'Annual' })
+        expect(optionBtn).toBeInTheDocument()
+      })
+
+      it('renders monthly option button as "selected"', async () => {
+        setup({ planValue: Plans.USERS_PR_INAPPM })
+        render(<ProPlanController {...props} />, { wrapper: wrapper() })
+
+        const optionBtn = await screen.findByRole('button', { name: 'Monthly' })
+        expect(optionBtn).toBeInTheDocument()
+        expect(optionBtn).toHaveClass('bg-ds-primary-base')
+      })
+
+      it.skip('renders the seat input with 10 seats (existing subscription)', async () => {
+        setup({ planValue: Plans.USERS_PR_INAPPM })
+        render(<ProPlanController {...props} />, { wrapper: wrapper() })
+
+        const seatCount = await screen.findByRole('spinbutton')
```

Comment:
This relates to the `input` tag in `TextInput`. With the way `<input>` is used, you can only check for `min` and `size` attributes on the tag. You could use those to assert on these tests by passing 10 as the `min` attribute and checking the input's min. 

Testing the value in the input box in TS is a bit [trickier](https://stackoverflow.com/questions/53071851/getting-the-value-from-input-element-in-typescript). You could try creating a state function that explicitly sets the value to access it easier if you feel strongly about testing it.

Reply:
Mmm issue isn't quite that, we can target the input's value, but that is determined by the `(...register('seats')` variable, which I'm unsure how to give a return value for

Reply:
Thinking about it, there's a test at the parent level that also does this and we don't need to mock this value, so I'll just move the test there

Reply:
The tests for that are already in the `UpgradeForm2` file, see this as an example, https://github.com/codecov/gazebo/blob/main/src/pages/PlanPage/subRoutes/UpgradePlanPage/UpgradeForm2/UpgradeForm2.spec.jsx#L655